App/Redux/PhotoViewingRedux.ts

```diff
@@ -4,21 +4,22 @@ import { ThreadId, Photo, PhotoId, Thread, ThreadName, Mnemonic } from '../Model
 
 const actions = {
   insertThread: createAction('INSERT_THREAD', (resolve) => {
-    return (thread: Thread) => resolve({ thread })
+    return (id: ThreadId, name: string) => resolve({ id, name })
   }),
   addThreadRequest: createAction('ADD_THREAD_REQUEST', (resolve) => {
-    return (name: ThreadName) => resolve({ name })
+    return (name: string, options?: { navigate?: boolean, sharePhoto?: { imageId: PhotoId, comment?: string } }) => resolve({ name }, options)
   }),
-  addThreadSuccess: createAction('ADD_THREAD_SUCCESS', (resolve) => {
-    return (thread: Thread) => resolve({ thread })
+  threadAdded: createAction('THREAD_ADDED', (resolve) => {
+    return (id: ThreadId, name: string) => resolve({ id, name })
   }),
   addThreadError: createAction('ADD_THREAD_ERROR', (resolve) => {
     return (error: any) => resolve({ error })
   }),
+  clearNewThreadActions: createAction('CLEAR_NEW_THREAD_ACTIONS'),
```

Comment:
This guy clears out the requests to navigate to or share a photo to the thread after it is created.